src/vsr/replica.zig

```diff
@@ -6299,43 +6461,26 @@ const DVCQuorum = struct {
         return log_view_max_.?;
     }
 
-    /// Returns the highest `commit_min` from any DVC (this is not a `commit_max`).
-    fn commit_min_max(dvc_quorum: QuorumMessages, local: struct {
-        replica: u64,
-        commit_min: u64,
-    }) u64 {
-        assert(dvc_quorum[local.replica].?.header.commit <= local.commit_min);
-
-        var commit_min_max_: ?u64 = null;
+    fn commit_max(dvc_quorum: QuorumMessages) u64 {
```

Comment:
Because the DVC generation now explicitly uses `commit_max` to anchor (i.e. to determine how many headers can be included) we must be more aggressive about advancing `commit_max` as much as possible.

Of the three ways we can do that:

```
            commit_max_ = std.math.max(commit_max_, dvc_commit_max_connected);
            commit_max_ = std.math.max(commit_max_, dvc_commit_max_pipeline);
            commit_max_ = std.math.max(commit_max_, dvc.header.commit);
```

Only the middle one (`dvc_commit_max_pipeline`) is new; the others were previously handled by `commit_min_max`.

The insight behind `dvc_commit_max_pipeline` is: if you have a pipeline-max of 4 ops, and the head op is 10, then op 6 _must_ be committed (somewhere).